From 465abd88f4f9477c72f0af8b683384912daf9f46 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Alex=20R=C3=B8nne=20Petersen?= <alex@alexrp.com>
Date: Thu, 18 Sep 2025 12:41:03 +0200
Subject: [PATCH 076/133] std.math.modf: disable vector test on LoongArch with
 LSX
X-Developer-Signature: v=1; a=openpgp-sha256; l=837; i=xtex@astrafall.org;
 h=from:subject; bh=MfecGy6PVgs5yWHENTzQR7dCASEf7jVrbAk6AhMj/qQ=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDJn78lhPnr1R2FP2XEhuXuJf/dPuGrvijeTf/7hQZFtw/
 UnS3EO7OyayMIhxMViKKbIUGTZ4s+qk84suK5eFmcPKBDJEWqSBgYGBgYWBLzcxr9RIx0jPVNtQ
 z9BQx0jHiIGLUwCm+rE7wx8ugQM8D1bPvPj0hk3lLWte3y8OJ9yuuyldr3h+5oF8op0yw1/p+V2
 P/HgvvnhQEKp52ciye3vb5Mf3n2cp3Ur9tejOprtMAA==
X-Developer-Key: i=xtex@astrafall.org; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

https://github.com/llvm/llvm-project/issues/159529
(cherry picked from commit 35d2b1e99a2bf57f76d3317061a1b74d145f3c59)
---
 lib/std/math/modf.zig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/std/math/modf.zig b/lib/std/math/modf.zig
index 870b4e6f6bd9..60680cf831a2 100644
--- a/lib/std/math/modf.zig
+++ b/lib/std/math/modf.zig
@@ -87,6 +87,7 @@ fn ModfTests(comptime T: type) type {
         test "vector" {
             if (builtin.os.tag == .macos and builtin.cpu.arch == .aarch64) return error.SkipZigTest;
             if (builtin.cpu.arch == .s390x) return error.SkipZigTest;
+            if (comptime builtin.cpu.has(.loongarch, .lsx)) return error.SkipZigTest; // https://github.com/llvm/llvm-project/issues/159529
 
             const widths = [_]comptime_int{ 1, 2, 3, 4, 8, 16 };
 
-- 
2.52.0

